Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -0,0 +1,198 @@
+    /// </example> 
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GcbqTable")]
+    public class GetGcbqTable : GcbqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        override public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset that you want to get a descriptor object for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset that you want to get a descriptor object for. 
+        /// (This will be used if both InputObject and Dataset are provided)
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        public Dataset InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset that you want to get a descriptor object for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 0)]
+        public string Table { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Select the DatasetID. Favor InputObject if both present. Default if neither present.
+            if (InputObject == null && Dataset == null)
+            {
+                ThrowTerminatingError(new ErrorRecord(
+                    new Exception("No Dataset Specified"),
+                    "No Dataset Specified",
+                    ErrorCategory.InvalidArgument,
+                    InputObject));
+                return;
+            }
+            Dataset = (InputObject != null) ? InputObject.DatasetReference.DatasetId : Dataset;
+
+            // Create and execute request.
+            if (Table == null)
+            {
+                TablesResource.ListRequest request = new TablesResource.ListRequest(Service, Project, Dataset);
+                var response = request.Execute();
+                if (response != null)
+                {
+                    WriteObject(response, true);
+                }
+                else
+                {
+                    WriteError(new ErrorRecord(
+                        new Exception("400"),
+                        "Error 400: List request to server failed.",
```

Comment:
I would avoid saying the request failed, or giving an http error code unless you actually get a `GoogleApiException`. The libraries we are using have a nasty tendency to have a null list when it should have an empty list. You can make a valid request, that should return no data, and get back `null`. I would add a verbose message to that effect. An error might lead users to kill their script even though the response was perfectly fine.